stacker_blueprints/generic.py

```diff
@@ -0,0 +1,92 @@
+from troposphere import (
+    Ref, Output
+)
+
+from stacker.blueprints.base import Blueprint
+from stacker.util import load_object_from_string
+
+
+class GenericResourceCreator(Blueprint):
+    """ Generic Blueprint for creating a resource """
+    def add_cfn_description(self):
+        """ Boilerplate for CFN Template """
+        template = self.template
+        template.add_version('2010-09-09')
+        template.add_description('Generic Resource Creator - 1.0.0')
+
+    """
+
+    *** NOTE ***    Template Version Reminder
+
+    Make Sure you bump up the template version number above if submitting
+    updates to the repo. This is the only way we can tell which version of
+    a template is in place on a running resouce.
+
+    """
+
+    VARIABLES = {
+        'Class':
+            {'type': str,
+             'description': 'The troposphere class to create, '
+                            'e.g.: ec2.Volume'},
+        'Output':
+            {'type': str,
+             'description': 'The output field that should be created, '
+                            'e.g.: VolumeId'},
+        'Properties':
+            {'type': dict,
+             'description': 'The list of properties to use for the '
+                            'Troposphere class'},
+    }
+
+    """
+
+    Example yaml:
+
+    - name: generic-resource-volume
+        class_path: blueprints.generic.GenericResourceCreator
+        variables:
+            Class: ec2.Volume
+            Output: VolumeId
+            Properties:
+                VolumeType: gp2
+                Size: 5
+                Encrypted: true
+                AvailabilityZone: us-east-1b
+
+    """
+
+    def setup_resource(self):
+        """ Setting Up Resource """
+        template = self.template
+        variables = self.get_variables()
+
+        tclass = variables['Class']
+        tprops = variables['Properties']
+        output = variables['Output']
+
+        klass = load_object_from_string('troposphere.' + tclass)
+
+        # we need to do the following because of type conversion issues -
+        # Troposphere expects strings, but stacker automatically converts
+        # to types such as Number, etc.  In the future, it'd be nice to
```

Comment:
Ahh, there seems to be a bug in the validator for `Size` on `ec2.Volume`, so that might be what is leading to your confusion:

https://github.com/cloudtools/troposphere/blob/master/troposphere/ec2.py#L469

My guess is that `Size` ought to have the integer validator.